src/clib/pio_rearrange.c

```diff
@@ -1589,6 +1632,23 @@ int box_rearrange_create_with_holes(iosystem_desc_t *ios, int maplen, const PIO_
         LOG((3, "iomaplen[%d] = %d", i, iomaplen[i]));
 #endif /* PIO_ENABLE_LOGGING */
 
+    /* Convert a 1-D index into a global coordinate value for each data element */
```

Comment:
Is it possible to re-use (refactoring if needed) idx_to_dim_list() to calculate gcoord_map (and possibly mark the function as a "static inline")?

Reply:
Also, if you are refactoring the function make sure that all arrays have an associated size specified as the function argument (The original code has not done a good job at this when using arrays)

Reply:
I do not like function calls inside a big loop so I manually inlined idx_to_dim_list().
inline-function is a concept of C++, not sure if it works for C code. Also, C++
compilers might choose not to inline a function as requested by users, especially
when there are loops inside that function like idx_to_dim_list().

Reply:
Using a function call might still be worth it, for readability etc, if it doesn't affect performance (Compilers are free to use/ignore inline , but unless we look at the assembly and see that the function is not inlined and results in significant performance overhead I think we shouldn't explicitly inline it) Can you try it (idx_to_dim_list() call ) out and measure the performance (with "-O2") with gnu/intel compilers?

Reply:
Your change might be getting the performance boost mostly due to moving the gcoord_map() calculation out of the loop (not inlining the function itself). Unless we measure the perf with the function call (outside the loop, like you have in your change) it would be hard to know.

Reply:
OK, should we also add "static inline" to coord_to_lindex()?

Reply:
> Also, if you are refactoring the function make sure that all arrays have an associated size specified as the function argument (The original code has not done a good job at this when using arrays)

Maybe we should do this in another PR later. idx_to_dim_list() is tested in test_rearr.c so
changing the function signature requires test changes.

Reply:
> OK, should we also add "static inline" to coord_to_lindex()?

Yes, go ahead and add "static inline" to both the functions in a single commit (one commit that just adds "static inline")

Reply:
> > Also, if you are refactoring the function make sure that all arrays have an associated size specified as the function argument (The original code has not done a good job at this when using arrays)
> 
> Maybe we should do this in another PR later. idx_to_dim_list() is tested in test_rearr.c so
> changing the function signature requires test changes.

Ok, we can change the function args (not critical) in a future PR.